web3/middleware/signing.py

```diff
@@ -0,0 +1,44 @@
+from web3.utils.transactions import (
+    fill_transaction_defaults
+)
+import eth_account
+from hexbytes import HexBytes
+
+from cytoolz import (
+    merge,
+)
+
+
+def key_to_account(web3, key):
+    if isinstance(key, eth_account.local.LocalAccount):
+        return key
+    if isinstance(key, (HexBytes, str, bytes)):
+        return web3.eth.account.privateKeyToAccount(key)
+    try:
+        # TODO Find a better way to check for and handle eth_keys objects
+        return web3.eth.account.privateKeyToAccount(key.to_hex())
+    except AttributeError:
+        raise ValueError("key must be one of the types (...). Was of type {0}".format(type(key)))
+
+
+def construct_in_flight_transaction_signing_middleware(private_key):
+
+    def in_flight_transaction_signing_middleware(make_request, web3):
```

Comment:
`local_singing_middleware`? (just looking for something a bit shorter)

Reply:
- `intercept_and_sign_middleware`
- `sign_and_send_raw_middleware`
- `fancy_pants_local_signing_middleware`